pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -51,18 +59,82 @@ var ManifestPath = "./bindata"
 // Add creates a new OperConfig Controller and adds it to the Manager. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) error {
-	return add(mgr, newReconciler(mgr, status, c))
+	rc, err := newReconciler(mgr, status, c)
+	if err != nil {
+		return err
+	}
+	return add(mgr, rc)
 }
 
 const ControllerName = "operconfig"
 
 // newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) *ReconcileOperConfig {
-	return &ReconcileOperConfig{
-		client: c,
-		status: status,
-		mapper: mgr.GetRESTMapper(),
+func newReconciler(mgr manager.Manager, status *statusmanager.StatusManager, c cnoclient.Client) (*ReconcileOperConfig, error) {
+	ctx := context.Background()
+	ctx, cancelFn := context.WithCancel(ctx)
+	stopCh := make(chan struct{})
+	go func() {
+		defer cancelFn()
+		<-stopCh
+	}()
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kube config: %v", err)
+	}
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create kube client: %w", err)
+	}
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	if err != nil {
+		return nil, err
 	}
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	desiredVersion := os.Getenv("RELEASE_VERSION")
+	missingVersion := "0.0.1-snapshot"
+
+	eventRecorder := events.NewKubeRecorder(kubeClient.CoreV1().Events("openshift-network-operator"), "cluster-network-operator", &corev1.ObjectReference{
+		APIVersion: "apps/v1",
+		Kind:       "Deployment",
+		Namespace:  "openshift-network-operator",
+		Name:       "network-operator",
+	})
+
+	// By default, this will exit(0) the process if the featuregates ever change to a different set of values.
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
+		eventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go configInformers.Start(stopCh)
```

Comment:
Have you considered using `c.Default().AddCustomInformer`?

Reply:
no i didn't even know it existed... lemme check

Reply:
pat and I spoke, this situation will end up in deadlock, the AddCustomInformer doesn't start until we create the newReconciler and this won't be created until the informer has started, so for now we go with goroutine approach.